packages/linux_can/lib/src/can_device.dart

```diff
@@ -311,6 +315,7 @@ class CanSocket implements Sink<CanFrame> {
   /// happen when sending lots of frames in a short time period. If [block] is false, this will throw a [LinuxError]
   /// with errno [EWOULDBLOCK] (value 22) in this case.
   Future<void> send(CanFrame frame, {bool block = true}) async {
+    assert(isFlexibleDataRate || frame is! CanFdFrame || frame is! CanFdFrameExtended);
```

Comment:
I would rather throw an exception here, in my mind asserts are just to verify internal invariants that should always hold. Some user of this API forgetting to check `isFlexibleDataRate` and sending a wrong frame can easily happen

```suggestion
    if (!isFlexibleDataRate && frame is CanFdFrame) {
      throw ArgumentError.value(frame, 'frame', 'CAN controller does not support CAN FD.');
    }
```